server/src/test/java/org/opensearch/index/remote/RemoteStoreStatsTrackerFactoryTests.java

```diff
@@ -0,0 +1,128 @@
+                updatedSize
+            )
+        );
+        assertEquals(defaultSize, remoteStoreStatsTrackerFactory.getMovingAverageWindowSize());
+    }
+
+    public void testUpdateMovingAverageWindowSizeLessThanMinAllowed() {
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard);
+        createSecondShard();
+
+        int defaultSize = RemoteStorePressureSettings.Defaults.MOVING_AVERAGE_WINDOW_SIZE;
+        assertDefaultSize(defaultSize);
+
+        int updatedSize = RemoteStorePressureSettings.Defaults.MOVING_AVERAGE_WINDOW_SIZE_MIN_VALUE - 1;
+        // TODO: The following exception isn't thrown currently
+        // assertThrows(
+        // IllegalArgumentException.class,
+        // () -> remoteStoreStatsTrackerFactory.updateMovingAverageWindowSize(
+        // RemoteSegmentTransferTracker::updateMovingAverageWindowSize,
+        // updatedSize
+        // )
+        // );
+        // assertEquals(defaultSize, remoteStoreStatsTrackerFactory.getMovingAverageWindowSize());
+        remoteStoreStatsTrackerFactory.updateMovingAverageWindowSize(
+            RemoteSegmentTransferTracker::updateMovingAverageWindowSize,
+            updatedSize
+        );
+        assertEquals(updatedSize, remoteStoreStatsTrackerFactory.getMovingAverageWindowSize());
+    }
+
+    public void testUpdateMovingAverageWindowSizeMinAllowed() {
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard);
+        createSecondShard();
+
+        int defaultSize = RemoteStorePressureSettings.Defaults.MOVING_AVERAGE_WINDOW_SIZE;
+        assertDefaultSize(defaultSize);
+
+        int updatedSize = RemoteStorePressureSettings.Defaults.MOVING_AVERAGE_WINDOW_SIZE_MIN_VALUE;
+        remoteStoreStatsTrackerFactory.updateMovingAverageWindowSize(
+            RemoteSegmentTransferTracker::updateMovingAverageWindowSize,
+            updatedSize
+        );
+        assertEquals(updatedSize, remoteStoreStatsTrackerFactory.getMovingAverageWindowSize());
+    }
+
+    private void assertDefaultSize(int defaultSize) {
+        assertEquals(defaultSize, remoteStoreStatsTrackerFactory.getMovingAverageWindowSize());
+        assertFalse(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId).isUploadBytesAverageReady());
+        assertFalse(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId).isUploadBytesPerSecAverageReady());
+        assertFalse(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId).isUploadTimeMsAverageReady());
+        if (shardId2 != null) {
+            assertFalse(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId2).isUploadBytesAverageReady());
+            assertFalse(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId2).isUploadBytesPerSecAverageReady());
+            assertFalse(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId2).isUploadTimeMsAverageReady());
+        }
+    }
```

Comment:
Pls check the https://github.com/opensearch-project/OpenSearch/blob/main/server/src/test/java/org/opensearch/index/remote/RemoteStorePressureSettingsTests.java and move these tests there. Some of the tests are there already, but lets add what might be absent.